src/object_cmd.cpp

```diff
@@ -228,6 +228,7 @@ CommandCost CmdBuildObject(TileIndex tile, DoCommandFlag flags, uint32 p1, uint3
 	if (type == OBJECT_OWNED_LAND) {
 		/* Owned land is special as it can be placed on any slope. */
 		cost.AddCost(DoCommand(tile, 0, 0, flags, CMD_LANDSCAPE_CLEAR));
+		cost.AddCost(EnsureNoShipFromDiagDirs(tile));
```

Comment:
Can this not use `AddCost` for any reason? Actually, shouldn't it check for ships before performing the Clear command above?

Reply:
I don't think the order matters that much, except for the error message priority. I think, if both fail, the 2nd error message is displayed. If only 1 fails, that error message is displayed and the command still fails, no matter the order.

The monetary cost is £0 for a success or fail on `EnsureNoShipFromDiagDirs`